server/src/main/java/org/opensearch/common/cache/tier/TieredCacheSpilloverStrategyService.java

```diff
@@ -193,6 +210,61 @@ private Function<K, CacheValue<V>> getValueFromTierCache(boolean trackStats) {
         };
     }
 
+    /**
+     * Dynamically add a new disk tier.
+     */
+    private void addNewDiskTier(DiskCachingTier<K, V> newTier) {
+        assert cachingTierList.size() == 1 && diskCachingTier.isEmpty() && !diskTierInUse;
+        // list only contains heap tier, and there is no inactive disk tier
+        diskCachingTier = Optional.of(newTier);
+        cachingTierList = List.of(onHeapCachingTier, diskCachingTier.get());
+        diskTierInUse = true;
+    }
+
+    /**
+     * Dynamically reenable an existing disk tier that has been disabled but not deleted.
+     */
+    private void enableInactiveDiskTier() {
+        assert cachingTierList.size() == 1 && diskCachingTier.isPresent() && !diskTierInUse;
+        // list only contains heap tier, and there is an inactive disk tier we can reactivate
+        cachingTierList = List.of(onHeapCachingTier, diskCachingTier.get());
+        diskTierInUse = true;
+    }
+
+    /**
+     * Dynamically disable an existing disk tier.
+     */
+    private void disableDiskTier() {
```

Comment:
i think we should return a boolean in these methods. Return false if something failed and true if succeeded. 
The caller can bubble it up as a failure or success. 
same for enableInactiveDiskTier, addNewDiskTier & toggleDiskTierEnabled
what do you think ?

Reply:
I like that idea